src/core/algorithms/od/set_based_verifier/verifier.cpp

```diff
@@ -0,0 +1,134 @@
+#include "verifier.h"
+
+#include <chrono>
+#include <stdexcept>
+
+#include <easylogging++.h>
+
+#include "algorithms/od/fastod/model/canonical_od.h"
+#include "config/column_index/option.h"
+#include "config/indices/option.h"
+#include "config/names_and_descriptions.h"
+#include "config/tabular_data/input_table/option.h"
+#include "util/range_to_string.h"
+#include "util/timed_invoke.h"
+
+namespace algos::od {
+
+SetBasedAodVerifier::SetBasedAodVerifier() : Algorithm({}) {
+    RegisterOptions();
+    MakeOptionsAvailable({config::kTableOpt.GetName()});
+}
+
+void SetBasedAodVerifier::RegisterOptions() {
+    using namespace config::names;
+    using namespace config::descriptions;
+    using config::Option;
+
+    auto get_cols_num = [this]() { return data_.GetColumnCount(); };
+    auto oc_before_set = [this]() {
+        if (!oc_.has_value()) {
+            oc_ = OC{};
+        }
+    };
+    auto ofd_before_set = [this]() {
+        if (!ofd_.has_value()) {
+            ofd_ = OFD{};
+        }
+    };
+    RegisterOption(config::kTableOpt(&input_table_));
+    RegisterOption(
+            config::IndicesOption{kOcContext, kDOcContext, true}(&oc_->context, get_cols_num)
+                    .SetBeforeSetCallback(oc_before_set)
+                    .SetConditionalOpts({{nullptr, {kOcLeftIndex, kOcRightIndex, kOcLeftOrdering}}})
+                    .SetIsRequiredFunc([this]() { return !ofd_.has_value(); }));
+    RegisterOption(
+            config::ColumnIndexOption{kOcLeftIndex, kDOcLeftIndex}(&oc_->left, get_cols_num));
+    RegisterOption(
+            config::ColumnIndexOption{kOcRightIndex, kDOcRightIndex}(&oc_->right, get_cols_num));
+    RegisterOption(Option<Ordering>{&oc_->left_ordering, kOcLeftOrdering, kDODLeftOrdering,
+                                    Ordering::ascending});
+    RegisterOption(
+            config::IndicesOption{kOFDContext, kDOFDContext, true}(&ofd_->context, get_cols_num)
+                    .SetBeforeSetCallback(ofd_before_set)
+                    .SetConditionalOpts({{nullptr, {kOFDRightIndex}}})
+                    .SetIsRequiredFunc([this]() { return !oc_.has_value(); }));
```

Comment:
IndicesOption regestiring looks very similar. Maybe we can extract it to a separate method, like `RegisterIndicesOption`

and use it like
```cpp
    RegisterIndicesOption(kOcContext, oc_, oc_before_set,
                          {kOcLeftIndex, kOcRightIndex, kOcLeftOrdering},
                          [this]() { return !ofd_.has_value(); });

    RegisterIndicesOption(kOFDContext, ofd_, ofd_before_set,
                          {kOFDRightIndex},
                          [this]() { return !oc_.has_value(); });
```